kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import java.util.Map;
+
+import io.delta.kernel.types.*;
+
+import io.delta.kernel.internal.actions.Metadata;
+
+/**
+ * Utilities related to the column mapping feature.
+ */
+public class ColumnMapping {
+    private ColumnMapping() {}
+
+    public static final String COLUMN_MAPPING_MODE_KEY = "delta.columnMapping.mode";
+    public static final String COLUMN_MAPPING_MODE_NONE = "none";
+    public static final String COLUMN_MAPPING_MODE_NAME = "name";
+    public static final String COLUMN_MAPPING_MODE_ID = "id";
+    public static final String COLUMN_MAPPING_PHYSICAL_NAME_KEY =
+        "delta.columnMapping.physicalName";
+    public static final String COLUMN_MAPPING_ID_KEY = "delta.columnMapping.id";
+
+    /**
+     * Returns the column mapping mode from the given configuration.
+     *
+     * @param  configuration Configuration
+     * @return Column mapping mode. One of ("none", "name", "id")
+     */
+    public static String getColumnMappingMode(Map<String, String> configuration) {
+        return configuration.getOrDefault(
+            COLUMN_MAPPING_MODE_KEY,
+            COLUMN_MAPPING_MODE_NONE);
+    }
+
+    /**
+     * Checks if the given column mapping mode in the given table metadata is supported.
+     *
+     * @param metadata Metadata of the table
+     */
+    public static void checkSupportedColumnMappingMode(Metadata metadata) {
```

Comment:
to me, `check___` doesn't imply that it throws.

`throwOnUnsupportedColumnMappingMode(metadata)`?